lib/rubocop/cop/rspec/repeated_example_group_description.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated example group descriptions.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'cool feature' do
+      #      # example group
+      #    end
+      #
+      #    describe 'cool feature' do
+      #      # example group
+      #    end
+      #
+      #    # bad
+      #    context 'when case x' do
+      #      # example group
+      #    end
+      #
+      #    describe 'when case x' do
+      #      # example group
+      #    end
+      #
+      #    # bad
```

Comment:
It's actually quite legit.

Reply:
@pirj so you say, that if there is repeated description, but have additional docstring or tags, it is ok and there should not be offenses, right?

Reply:
Also I've checked similar cop `RSpec/RepeatedDescription` for example and I see offenses for such kind of piece of code
```ruby
it 'do something', :flag do
  # ...
end

it 'do something', :flag do
  # ...
end
```

Maybe I'm missing something, didn't understand your point fully or we have different rules for example groups

Reply:
I can't find a good real example, but I remember seeing some this in [`real-world-ruby-apps`](https://github.com/jeromedalbert/real-world-ruby-apps/) or [`real-world-rails`](https://github.com/eliotsykes/real-world-rails).

RSpec maintainers consider additional docstrings as a feature that should fade away https://github.com/rspec/rspec-core/pull/2681#discussion_r361811453

However, I'd stay strong for metadata, sometimes it's even used to construct examples/example groups. This metadata is so descriptive that it doesn't make sense to change the title, e.g.:
```ruby
it 'runs pretty quickly', load_factor: 20 do
end

it 'runs pretty quickly', load_factor: 100 do
end
```
I see those as two distinct examples and see no good reason to flag them as repeated.
The same applies to similar bodies. Even though they might be exactly the same, it doesn't mean there's repetition that should by DRY'd out.

For the example groups, specifically:
```ruby
context group: :guest do
  it ...
end
```
actually constructs a wrapping context with a generated docstring `when the group is "guest"`, and it makes it distinct from the other one.

Using metadata might also be seen similar to passing values to shared examples, you wouldn't consider two calls if `it_behaves_like` with different parameters repetition, would you?

`RepeatedDescription` does not take metadata into account, and I see it as a flaw, not as an intended design.

Anyway, I'm all for discussing what are your rules for example groups.

Reply:
@pirj Ok, I got it, thanks for the detailed answer, I will fix it then